libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2415,6 +2424,8 @@ void WaveTrack::GetEnvelopeValues(double *buffer, size_t bufferLen,
          clip->GetEnvelope()->GetValues(rbuf, rlen, rt0, tstep);
       }
    }
+   if (backwards)
+      std::reverse(buffer, buffer + bufferLen);
```

Comment:
I see reversal of the envelope values.  But I was expecting to see `ReverseSamples` too, restoring the removed call in MixerSource.

I think this means the first commit breaks backward scrubbing, and that should not be.  I haven't yet reviewed later commits, which might make it work again, when the WaveTrack has its stretching decorator, but logically, shouldn't the proper place to apply reversal be here in WaveTrack?

Reply:
The stretching decorator re-implements ::Get, where the clip samples are reversed before being passed to the stretching algorithm. As a result scrubbing does work when injected that decorator, but you're right, `WaveTrack::Get` should reverse the samples, even if no other code has use for it. `WaveTrack::GetSampleView` isn't a virtual function and `backwards` could/should be removed from its arguments, leaving `CachingPlayableSequence::Get` do the reversal.

Reply:
In the interests of progress I will not block the PR if this is left unfinished, and scrubbing works in the end result, but you are aware of a need to come back and clean it up.

Really you should make sure that the sample reversal is part of the stated contract of `WideSampleSequence` and that all overrides of `Get` do it correctly -- whether it's the undecorated or the decorated WaveTrack.  Maybe some overrides of `Get` won't be directly called in 3.4, but you should not assume it will always be so.  The classes should make sense in isolation and in test harnesses, fulfilling the base class contract.  That's what the L of SOLID is all about.

Reply:
> WaveTrack::Get should reverse the samples

I meant by that that I was going to address it. It's done now.